build.ps1

```diff
@@ -643,23 +611,70 @@ function Build-XCTest($Arch)
+function Consolidate-RuntimeInstall($Arch)
+{
+  $RuntimeInstallRoot = Get-RuntimeInstallDir $Arch
+  Remove-Item -Force -Recurse $RuntimeInstallRoot -ErrorAction Ignore
+  
+  $BinariesSrc = Get-ArchSpecificPlatformDir $Arch "$PlatformSDKSubDir\usr\bin"
+  $BinariesDst = "$RuntimeInstallRoot\usr\bin"
+  New-Item -ItemType Directory $BinariesDst | Out-Null
+  Copy-Item $BinariesSrc\* $BinariesDst\
+}
 
-  # Restructure Module
-  New-Item -ErrorAction Ignore -Type Directory `
-    -Path $InstallDir\lib\swift\windows\XCTest.swiftmodule
-  Move-Item -Force `
-    $InstallDir\lib\swift\windows\$($Arch.LLVMName)\XCTest.swiftdoc `
-    $InstallDir\lib\swift\windows\XCTest.swiftmodule\$($Arch.LLVMTarget).swiftdoc
-  Move-Item -Force `
-    $InstallDir\lib\swift\windows\$($Arch.LLVMName)\XCTest.swiftmodule `
-    $InstallDir\lib\swift\windows\XCTest.swiftmodule\$($Arch.LLVMTarget).swiftmodule
+# Copies files installed by CMake from the arch-specific platform root,
+# where they follow the layout expected by the installer,
+# to the final platform root, following the installer layout.
+function Consolidate-PlatformInstall($Arch)
+{
+  Remove-Item -Force -Recurse $PlatformInstallRoot -ErrorAction Ignore
+
+  $ArchSpecificPlatformInstallRoot = Get-ArchSpecificPlatformDir $Arch
+  $ArchSpecificSDKInstallRoot = "$ArchSpecificPlatformInstallRoot\$PlatformSDKSubDir"
+
+  # Copy SDK header files
+  $IncludeSrc = "$ArchSpecificSDKInstallRoot\usr\include"
+  $LibIncludeSrc = "$ArchSpecificSDKInstallRoot\usr\lib\swift"
+  $IncludeDst = "$SDKInstallRoot\usr\include"
+  Copy-Item -Force -Recurse -ErrorAction Ignore $IncludeSrc $IncludeDst
+  Get-ChildItem -Recurse -Filter *.h $LibIncludeSrc | ForEach-Object {
+    $DstPath = $_.FullName.Replace($LibIncludeSrc, $IncludeDst)
+    New-Item -Type Directory -ErrorAction Ignore ([IO.Path]::GetDirectoryName($DstPath)) | Out-Null
+    Copy-Item -Force $_.FullName $DstPath # Overwrites the file contents
+  }
+
+  # Remove CoreFoundation headers
+  foreach ($Module in ("CoreFoundation", "CFXMLInterface", "CFURLSessionInterface"))
+  {
+    Remove-Item -Recurse -Force -ErrorAction Ignore `
+      $IncludeDst\$Module
+  }
+
+  # Copy SDK libs, placing them in an arch-specific directory
+  $WindowsLibSrc = "$ArchSpecificSDKInstallRoot\usr\lib\swift\windows"
+  $WindowsLibDst = "$SDKInstallRoot\usr\lib\swift\windows"
+
+  New-Item -Type Directory $WindowsLibDst\$($Arch.LLVMName) | Out-Null
+  Copy-Item -Force $WindowsLibSrc\*.lib $WindowsLibDst\$($Arch.LLVMName)
+
+  # Copy SDK modules, restructuring to the installed layout
+  Get-ChildItem -Recurse -Filter *.swiftmodule $WindowsLibSrc\$($Arch.LLVMName) | ForEach-Object {
+    $DstDir = "$WindowsLibDst\$($_.Name)"
+    New-Item -ItemType Directory $DstDir | Out-Null
+
+    # Copy all module files, renaming to the arch-specific variant
+    Get-ChildItem $_.Directory | ForEach-Object {
+      Copy-Item -Force $_.FullName "$DstDir\$($Arch.LLVMTarget)$($_.Extension)"
+    }
+  }
```

Comment:
Doesn't this prevent us from actually using the SDK during the build?  I'd like to be able to use the SDK layout during the build and cross-compile.

Reply:
I iterated until it diffs to the same files as the installed sdk. I'm going to do a clean build overnight to confirm this

Reply:
Final tweaks this morning and I could use the SDK to build every other project up to and including sourcekit-lsp